extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/connection/ConnectionService.java

```diff
@@ -167,18 +201,34 @@
                             // Parse name
                             String name = parameter.substring(0, equals);
                             String value = parameter.substring(equals+1);
-
-                            config.setParameter(name, value);
+                            
+                            // Pull out and set proxy parameters, if present
+                            // Otherwise set the parameter.
+                            switch(name) {
+                                case PROXY_HOST_PARAMETER:
```

Comment:
These should probably be separate, dedicated LDAP attributes, rather than reusing `guacConfigParameter` (which is meant for connection parameters). This would mean changes to the LDAP schema, which would mean new OID values.

I have the OID number assignment scheme written down on a scrap of paper from years past ... I'll locate my notes and copy them here. There's an OID space which was reserved for Guacamole ages ago (see 38971 within https://www.iana.org/assignments/enterprise-numbers/enterprise-numbers), but there is a system for consistently assigning numbers within that space.

Reply:
Sounds good.  I hesitated to modify the LDAP schema, but I'm happy to go that route if that's the best way to go.  Do you think separate LDAP attributes specifically for proxying, or just a generic Guacamole Attribute that is handled similar to the parameters?

For the OID, should we use the Guacamole one, or does ASF have one we should go with?

Reply:
We could switch over to the ASF one: https://cwiki.apache.org/confluence/display/DIRxPMGT/OID+Assignment+Scheme

Probably a good idea to do so. I'll still locate the ancient scrap of paper to server as a reference for assigning numbers from that, though - we need to use some consistent method for assigning the numbers within the Guacamole part of the ASF space (once that exists).

Reply:
Cool.  Do we need to open an INFRA ticket to get that OID assigned under ASF?  Or is there some documentation somewhere on the ASF site that documents this?

Reply:
According to https://cwiki.apache.org/confluence/display/DIRxPMGT/OID+Assignment+Scheme:

> Each contact person is the authority for assigning unique OID values and ranges to projects or persons. Contact that person for more assignments.

The contact person listed for the main ASF OID is Alex Karasulu. Feels weird to me to not use a mailing list, but I'll shoot him an email. ;)

Reply:
Depending on OID assignment, it looks like we should be fine with the above system. There is no standard limit imposed on OID length, and longer OIDs are already actively used.

Reply:
Cool.  I'm thinking the right way to do this is to continue with the current OID in this PR, and then open a separate JIRA issue (and PR) for converting from the old OID to the ASF OID?

Reply:
Okay, created a new LDAP attribute for the proxy configuration, under the existing OID string.

Reply:
Alrighty - we now have confirmation from the Apache Directory folks: https://lists.apache.org/thread.html/1897a20401348da14c4b20483464ee8ee06ffb510a29fa86d475d200@%3Cdev.directory.apache.org%3E

We can proceed with `1.3.6.1.4.1.18060.18`, and they will be updating the Apache Directory website to list the various reserved ASF OID branches.

Reply:
Cool.  Would you rather go ahead and make those changes in this PR, or go the route of separating that into another issue (I'd already opened a separate JIRA issue for it...)?

Reply:
I think changing OIDs will be saved for a different JIRA issue...